Mergin/project_status_dialog.py

```diff
@@ -204,8 +203,13 @@ def show_changes(self):
         if not self.changes_summary:
             self.ui.messageBar.pushMessage("Mergin", "No changes found in the project layers.", Qgis.Info)
             return
-
-        self.close()
+        check_result = unsaved_project_check()
+        if check_result == UnsavedChangesStrategy.HasUnsavedChanges:
+            return
```

Comment:
Not sure if returning is the right solution here. Let's imagine situation that some changes were made and saved, while some are not saved. Probably we still should show viewer with the warning in that case.

I'm not an UX expert, and would welcome second opinion here.

Reply:
As far as I understand from the code, this would open the "unsaved changes" dialog again, and only return prematurely if "Cancel" was clicked. This logic seems fine to me...